packages/fxa-settings/src/models/Account.ts

```diff
@@ -1046,4 +1050,21 @@ export class Account implements AccountData {
     firefox.accountDeleted(this.uid);
     Storage.factory('localStorage').clear();
   }
+
+  async resetPasswordWithRecoveryKey(opts:{
+    accountResetToken:string,
+    emailToHashWith:string
+    password:string,
+    recoveryKeyId:string,
+    kB:string,
+  }) {
+    const data = await this.authClient.resetPasswordWithRecoveryKey(
```

Comment:
We should be able to use the gql password reset mutation here. Or you can do in a follow up since PR is already kind of big.

Reply:
I didn't see an implementation that supported the recovery key. Did I just overlook it? If it needs to be implemented, it seems like it should be a separate ticket maybe?